cmd/podman/shared/pod.go

```diff
@@ -134,4 +137,129 @@ func CreatePortBindings(ports []string) ([]ocicni.PortMapping, error) {
 	return portBindings, nil
 }
 
+func GetPodsWithFilters(r *libpod.Runtime, c *cliconfig.PodPsValues) ([]*libpod.Pod, error) {
+
+	var pods []*libpod.Pod
+	var err error
+	if c.Latest {
+		pod, err := r.GetLatestPod()
+		if err != nil {
+			return nil, err
+		}
+		pods = append(pods, pod)
+	} else {
+		pods, err = GenerateFilterFunctionAndGetPods(r, strings.Split(c.Filter, ","))
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return pods, nil
+}
+
+func GenerateFilterFunctionAndGetPods(r *libpod.Runtime, filters []string) ([]*libpod.Pod, error) {
+	var filterFuncs []libpod.PodFilter
+	for _, f := range filters {
+		filterSplit := strings.Split(f, "=")
+		if len(filterSplit) < 2 {
+			return nil, errors.Errorf("filter input must be in the form of filter=value: %s is invalid", f)
+		}
+		generatedFunc, err := GeneratePodFilterFuncs(filterSplit[0], filterSplit[1])
+		if err != nil {
+			return nil, errors.Wrapf(err, "invalid filter")
+		}
+		filterFuncs = append(filterFuncs, generatedFunc)
+	}
+	pods, err := r.Pods(filterFuncs...)
+	if err != nil {
+		return nil, err
+	}
+
+	return pods, nil
+}
+func GeneratePodFilterFuncs(filter, filterValue string) (
```

Comment:
As above - function comment, and should this be public?

Reply:
fixed. made private.